staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -128,17 +127,22 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 		}
 		trace.Step("Conversion done")
 
+		// On create, get name from new object if unset
+		if len(name) == 0 {
+			_, name, _ = scope.Namer.ObjectName(obj)
+
+			if gvk.Kind == "Namespace" {
+				namespace = name
+			}
```

Comment:
1. compare the full group/kind
2. shouldn't this be outside the `if len(name) == 0` block?

Reply:
Thanks for spotting it, you are right, I will change both points.